### ISSUE TITLE
Update docs for v0.14.0

### ISSUE THREAD

**Issue by JordanMartinez** [purescript/purescript-unsafe-coerce#10]
This library exports `unsafeCoerce`, which has the following docs. Now that we have `safe-coerce`, what should the example be in these docs for why one would use it (i.e. paragraphs 2-4)?

```
-- | A _highly unsafe_ function, which can be used to persuade the type system that
-- | any type is the same as any other type. When using this function, it is your
-- | (that is, the caller's) responsibility to ensure that the underlying
-- | representation for both types is the same.
-- |
-- | One application for this function is to avoid doing work that you know is a
-- | no-op because of newtypes. For example, if you have an `Array (Conj a)` and you
-- | want an `Array (Disj a)`, you could do `Data.Array.map (runConj >>> Disj)`, but
-- | this performs an unnecessary traversal. `unsafeCoerce` accomplishes the same
-- | for free.
-- |
-- | It is highly recommended to define specializations of this function rather than
-- | using it as-is. For example:
-- |
-- | ```purescript
-- | mapConjToDisj :: forall a. Array (Conj a) -> Array (Disj a)
-- | mapConjToDisj = unsafeCoerce
-- | ```
-- |
-- | This way, you won't have any nasty surprises due to the inferred type being
-- | different to what you expected.
```

**Comment by JordanMartinez** [comment]
Otherwise, this library compiles fine.

**Comment by garyb** [comment]
The example it uses with `Array (Conj a)` is actually a great example of what safe coerce is meant for now :smile:. I'd remove that from the comment here, since safe coercions should be encouraged wherever they work.

In the presence of `Coercible` I'd say there's no simple example that can be provided here, since `unsafeCoerce` should only be used to do things that can't be expressed within PureScript itself. I think just taking that "One application..." paragraph out would suffice though, and maybe replacing it with a line directing people to use safe-coerce unless they really know what they are doing.

**Comment by hdgarrood** [comment]
Some examples I can think of where you might still want to use unsafeCoerce:

* mimicking existentials:  https://discourse.purescript.org/t/data-declaration-using-row-type/1024/2
* FFI boundaries where some function from a JS library can take values of different types as its argument, so you define a `foreign import data` type for the argument and provide ‘construction’ functions whose implementation is unsafeCoerce just for the types where it’s valid.

**Comment by JordanMartinez** [comment]
> The example it uses with Array (Conj a) is actually a great example of what safe coerce is meant for now smile. I'd remove that from the comment here, since safe coercions should be encouraged wherever they work.

If it's not already there, I think this doc comment should be moved to that repo instead.

**Comment by hdgarrood** [comment]
Resolved by #11
